src/utils/BlockUtils.ts

```diff
@@ -0,0 +1,52 @@
+import type { Provider } from "@ethersproject/abstract-provider";
+import { isDefined } from "./TypeGuards";
+import { getCurrentTime } from "./TimeUtils";
+
+type Opts = {
+  latestBlockNumber?: number;
+  blockRange?: number;
+};
+
+type BlockTimeAverage = {
+  average: number;
+  blockRange: number;
+  timestamp: number;
+};
+
+// Archive requests typically commence at 128 blocks past the head of the chain.
+// Round down to 120 blocks to avoid slipping into archive territory.
+const defaultBlockRange = 120;
+
+// Retain computations for 15 minutes.
+const cacheTTL = 60 * 15;
+const blockTimes: { [chainId: number]: BlockTimeAverage } = {};
+
+/**
+ * @description Compute the average block time over a block range.
+ * @returns Average number of seconds per block.
+ */
+export async function averageBlockTime(
```

Comment:
Is this directly copied or is there a section here you. need further reviews on?

Reply:
This is all new code, but it's inspired by the interface of `averageBlockTimeSeconds()` from UMA/sdk. So I think it'd be good with a sanity check on the implementation.

https://github.com/UMAprotocol/protocol/blob/a80c0e218bd48c9a3c714c96fe25877e329a3c0a/packages/sdk/src/utils.ts#L103.